File: server/ruby/app/infrastructure/mappers/payment_details/card_fraud_results_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class CardFraudResultsMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::CardFraudResults.new
          dto.avs_result = response.avs_result
          dto.fraud_service_result = response.fraud_service_result
          dto.cvv_result = response.cvv_result

          dto
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/card_fraud_results_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.card_fraud_results import CardFraudResults
from onlinepayments.sdk.domain.card_fraud_results import CardFraudResults as CardFraudResultsSdk

class CardFraudResultsMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[CardFraudResultsSdk]) -> Optional[CardFraudResults]:
        if response is None:
            return None

        dto = CardFraudResults()
        dto.avs_result = response.avs_result
        dto.fraud_service_result = response.fraud_service_result
        dto.cvv_result = response.cvv_result

        return dto